### PR TITLE
Add diagnostics

### PR DESCRIPTION
**Pull request by raphaelsaavedra**
Addressing #99

### CODE REVIEW

**Comment by raphaelsaavedra** [comment]
Later on, need to add diagnostics to the README, docs and examples

File: src/diagnostics.jl [hunk 1]
```diff
@@ -0,0 +1,92 @@
+"""Run diagnostics and print results"""
```
Comment (guilhermebodin) [inline src/diagnostics.jl]: change all function descriptions to this template
"""
    diagnostics(ss::StateSpace)

Run diagnostics and print results
"""

File: src/diagnostics.jl [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        println("                  Ljung-Box: p-value = $(round(pvalueLB[j], digits = 5))                 ")
+        println("           Homoscedasticity: p-value = $(round(pvalueH[j], digits = 5))              ")
+    end
+
+    println("==============================================================")
+
+    return Diagnostics(pvalueJB, pvalueLB, pvalueH)
+end
+
+"""Obtain standardized residuals from a state-space model"""
```
Comment (guilhermebodin) [inline src/diagnostics.jl]: here too

File: src/diagnostics.jl [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+            # Replace Inf with large numbers to avoid numerical issues
+            e[t, j] == Inf && (e[t, j] = 1e6)
+            e[t, j] == -Inf && (e[t, j] = -1e6)
+        end
+    end
+    return e
+end
+
+"""Run Jarque-Bera normality test and return p-values"""
```
Comment (guilhermebodin) [inline src/diagnostics.jl]: here too

File: src/diagnostics.jl [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        statistic = n * b1/6 + n*(b2 - 3)^2/24
+        d = Chisq(2.)
+        pvalue[j] = 1. - cdf(d, statistic)
+    end
+    return pvalue
+end
+
+"""Run Ljung-Box independence test and return p-values"""
+function ljungbox(e::Matrix{Float64}; maxlag = 20)
+    n = size(e, 1)
```
Comment (guilhermebodin) [inline src/diagnostics.jl]: `(n , p) = size(e)`

**Comment by raphaelsaavedra** [comment]
The docs can certainly still be improved (including lots of stuff other than diagnostics), but I have added some mentions to the diagnostics in the docs and also some tests in order to improve coverage

File: test/test_results.jl [hunk 5]
```diff
@@ -1,5 +1,5 @@
 AP = CSV.read("../examples/AirPassengers.csv")
-logAP = log.(Vector{Float64}(AP[:Passengers]))
+logAP = log.(Vector{Float64}(AP[!, :Passengers]))
```
Comment (guilhermebodin) [inline test/test_results.jl]: what does this change?

Comment (raphaelsaavedra) [reply test/test_results.jl]: Nothing, it's to remove a warning saying the previous syntax was deprecated

**Pull request merged**